src/components/list/SearchableList.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+
+import Fuse, { FuseResult } from 'fuse.js';
+
+import { SearchField, SearchFieldOptions } from '../input/SearchField';
+import Flex from '../flex';
+import { ListItemType, SearchableListItem } from './SearchableListItem';
+
+import List from './';
+
+export type ListItemWithChildren = ListItemType & {
+  children?: Array<ListItemType>;
+};
+
+export type Props = {
+  listItems: Array<ListItemWithChildren>;
+  noResults?: FC;
+  insertion?: FC;
```

Comment:
Struggled to come up with a name here, the thing that goes in between the search filed and options in case the query is empty

![Screenshot 2024-09-04 at 15 26 39](https://github.com/user-attachments/assets/a95fcc7e-6aa8-465f-94aa-9b6c66ff3182)

Reply:
Why not just `children`?

Reply:
It feels kind of off in terms of a standard component taking children renders them in an enriched way (eg: Adding formatting on a text) whereas this one fills a specific behaviour within the element to be displayed when there is no search (my initial name was placeholder)

Reply:
Yes, you're right I wouldn't expect children to disappear conditionally. What about `emptyQueryContent`/`Extra`? Or something in that direction?